idseq_dag/util/command.py

```diff
@@ -186,25 +189,35 @@ def subprocess_scope(*args, **kwargs):
     return wrapper
 
 
-def retry(operation, randgen=random.Random().random):
+def retry(operation, MAX_TRIES=3):
     """Retry decorator for external commands."""
     # Note the use of a separate random generator for retries so transient
     # errors won't perturb other random streams used in the application.
+    invocation = [0]
```

Comment:
I am a bit confused why this is a list. I have seen cases where this is required for updating in certain scenarios but I am not sure why that applies here. Why is it defined outside the function when it doesn't appear to be referenced outside? Could you explain in a comment?

Reply:
I believe it is required for this exact reason.   Otherwise, if in the nest function you just did `invocation = invocation + 1` it would actually create a new local variable called `invocation`.   That's python.

Reply:
I will add a comment to that end.